magda-web-client/src/UI/DataPreviewVis.js

```diff
@@ -0,0 +1,123 @@
+import 'es6-symbol/implement';
+import React, { Component } from 'react';
+import Tabs from 'muicss/lib/react/tabs';
+import Tab from 'muicss/lib/react/tab';
+import ChartConfig from './ChartConfig';
+import defined from '../helpers/defined';
+import {fetchPreviewData} from '../actions/previewDataActions';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import VegaLite from 'react-vega-lite';
+import DataPreviewTable from './DataPreviewTable';
+import './DataPreviewVis.css';
+
+class DataPreviewVis extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chartType: 'line',
+      chartTitle: '',
+      Yaxis: '',
+      xAxis: '',
+      xScale: 'temporal',
+      yScale: 'quantitative',
+    }
+  }
+
+  componentWillMount(){
+    if(this.props.distribution){
+      this.props.fetchPreviewData(this.props.distribution);
+    }
+  }
+
+  componentWillReceiveProps(nextProps){
+    if(nextProps.distribution.identifier !== this.props.distribution.identifier){
+        this.props.fetchPreviewData(nextProps.distribution);
+      }
+  }
+
+  onChange(i, value, tab, ev) {
```

Comment:
Maybe remove these if they don't do anything?

Reply:
I think it should be covered once I merge the complete chart section into it.